### ISSUE TITLE
unittest flake: TestControllerSync Warning ClaimLost Bound claim has lost its PV

### ISSUE THREAD

**Issue by bprashanth** [kubernetes/kubernetes#26578]
https://pantheon.corp.google.com/storage/browser/kubernetes-jenkins/pr-logs/pull/26335/kubernetes-pull-test-unit-integration/28566/artifacts/
```
--- FAIL: TestControllerSync (0.13s)
	framework_test.go:395: Unexpected event received: "Warning ClaimLost Bound claim has lost its PersistentVolume. Data on the volume is lost!"
	framework_test.go:739: Test "5-4 - delete volume": Events do not match
```
@kubernetes/sig-storage @jsafrane please close if it's fixed by https://github.com/kubernetes/kubernetes/pull/26555

**Comment by mikedanese** [comment]
This is failing a bunch

**Comment by mikedanese** [comment]
I've seen similar test failures waiting on events in other controller tests so i'll point out that events are generated async:

https://github.com/kubernetes/kubernetes/blob/master/pkg/client/record/event.go#L272

I have no idea if this is the root cause but it doesn't seem to be accounted for in framework_test.go.

**Comment by jsafrane** [comment]
Indeed, the unit test stopped stopped recording events too early. Fixed in #26619.
